RIOT-OS/RIOT
# sys/crypto: Add PKCS#7 padding support to CBC mode encryption

### Contribution description
This extends the CBC mode API with functions for CBC with padding, allowing for plaintext input that is not a multiple of the AES block size.
It implements PKCS#7 padding.

### Testing procedure
There are also additional positive and negative test cases to verify the implementation in the tests/sys_crypto directory.
Just run the tests as usual.

### Issues/PRs references
This is a new version of PR #7900
